src/Sylius/Bundle/ShopBundle/EventListener/OrderPromotionIntegrityChecker.php

```diff
@@ -87,10 +74,22 @@ public function check(ResourceControllerEvent $event): void
 
                 break;
```

Comment:
Have someone though about what will happen when there are several promotions that are no applicable? 😄
My guess is that it will remove all, but notify only about the first.